geoextent/lib/helpfunctions.py

```diff
@@ -98,32 +97,53 @@ def getDelimiter(csv_file):
     csv_file.seek(0)
     return csv.reader(csv_file.readlines(), delimiter=dialect.delimiter)
 
-def date_parser_iso8601(list):
+def get_time_format(time_list):
     '''
-    Function purpose: help-function to transform tbox from extracted time format into ISO8601
-    Input: list (tbox) \n
-    Output: array values
+    Function purpose: 'Guess' time format of a list of 'strings' by taking a representative sample
+    Input: list of strings \n
+    Output: time format in string format (e.g '%Y.%M.d')
     '''
 
-    time_1, time_2 = None, None
-    time_form = [time_format(np.array([list[0]])), time_format(np.array([list[1]]))]
-
-    if time_form[0] == time_form[1]:
-        time_1 = pd.to_datetime(list[0], format=time_form[0])
-        time_2 = pd.to_datetime(list[1], format=time_form[0])
+    date_time_format = None
+    num_sample = 30
 
+    if len(time_list) <= num_sample:
+        time_sample = time_list
     else:
-        for i in time_form:
+        time_sample = random.sample(time_list, 30)
+
+    format_list = []
+
+    for i in range(0, len(time_sample)):
+        format_list.append(time_format(np.array([time_sample[i]])))
+
+    unique_formats = list(set(format_list))
+
+    if unique_formats is not None:
+        for tf in unique_formats:
             try:
-                time_1 = pd.to_datetime(list[0], format=i)
-                time_2 = pd.to_datetime(list[1], format=i)
+                pd.to_datetime(time_sample, format=tf)
+                date_time_format = tf
             except:
                 pass
+    else:
+        return None
+
+    return date_time_format
 
-    if time_1 is not None and time_2 is not None:
-        time_1 = time_1.strftime('%Y-%m-%d')
-        time_2 = time_2.strftime('%Y-%m-%d')
 
-    bbox_iso8601 = [time_1, time_2]
+def date_parser(datetime_list):
+    '''
+    Function purpose: transform list of strings into ISO8601 ('%Y-%m-%d')
```

Comment:
Please update the user documentation to make clear that the temporal extent is only at "day" precision.

Reply:
OK ! I'll update documentation for CLI and API adding the "day" precision.